ts-jest not reading types from my declaration files

## 🐛 Bug Report

<!-- A clear and concise description of what the bug is. -->

Interfaces defined in my project's `.d.ts` file are not picked up by `ts-jest`.

My project uses types defined in `plugin/types/interface.d.ts`. The source code in `plugin/src` references these types and compiles without issue. But if these types are referenced within the context of `ts-jest`, the test suite fails to run with _Cannot find name_ errors.

## To Reproduce

The complete test setup can be seen [in this draft PR](https://github.com/gatsbyjs/gatsby-source-shopify-experimental/pull/73/files)

Steps to reproduce the behavior:

1. Clone repo: git@github.com:gatsbyjs/gatsby-source-shopify-experimental.git
2. Checkout branch sslotsky/test-setup
3. Install dependencies with `yarn`
4. Switch into the `plugin/` directory
5. Run `yarn test`

## Expected behavior

<!-- A clear and concise description of what you expected to happen. -->

The tests should pass, but instead:

```
 FAIL  test/sample-test.spec.ts
  ● Test suite failed to run

    src/client.ts:3:28 - error TS2304: Cannot find name 'ShopifyPluginOptions'.

    3 const adminUrl = (options: ShopifyPluginOptions) =>
                                 ~~~~~~~~~~~~~~~~~~~~
    src/client.ts:8:39 - error TS2304: Cannot find name 'ShopifyPluginOptions'.

```

Note that VSCode is able to find the interface in the screenshot below, which tells me that from a TypeScript perspective, my configuration _should_ be ok, because otherwise VSCode shows me lots of type errors if I have those configurations wrong.

![image](https://user-images.githubusercontent.com/374078/110819444-d0170c80-8253-11eb-80ce-ed8fe668769d.png)


## Link to repo (highly encouraged)

https://github.com/gatsbyjs/gatsby-source-shopify-experimental/pull/73/files

<!--
Please provide either a minimal repository either on GitHub or GitLab.
Issues without a reproduction link are likely to stall.
-->

## Debug log:
<!-- You can activate the debug logger by setting the environment variable TS_JEST_LOG=ts-jest.log before running tests. 
The output of the logger will be in **ts-jest.log** in CWD, paste it in the below section: 
-->

```
# content of ts-jest.log :
{"context":{"allowJs":false,"logLevel":20,"namespace":"jest-preset","package":"ts-jest","version":"26.5.3"},"message":"creating jest presets not handling JavaScript files","sequence":1,"time":"2021-03-11T16:29:22.994Z"}
{"context":{"allowJs":false,"logLevel":20,"namespace":"jest-preset","package":"ts-jest","version":"26.5.3"},"message":"creating jest presets not handling JavaScript files","sequence":2,"time":"2021-03-11T16:29:23.024Z"}
{"context":{"logLevel":20,"namespace":"Importer","package":"ts-jest","version":"26.5.3"},"message":"creating Importer singleton","sequence":3,"time":"2021-03-11T16:29:23.573Z"}
{"context":{"actualVersion":"26.6.3","expectedVersion":">=26 <27","logLevel":20,"namespace":"versions","package":"ts-jest","version":"26.5.3"},"message":"checking version of jest: OK","sequence":4,"time":"2021-03-11T16:29:23.578Z"}
{"context":{"logLevel":20,"namespace":"ts-jest-transformer","package":"ts-jest","version":"26.5.3"},"message":"created new transformer","sequence":5,"time":"2021-03-11T16:29:23.578Z"}
{"context":{"logLevel":30,"namespace":"ts-jest-transformer","package":"ts-jest","version":"26.5.3"},"message":"no matching config-set found, creating a new one","sequence":6,"time":"2021-03-11T16:29:23.579Z"}
{"context":{"config":{"automock":false,"cache":true,"cacheDirectory":"/tmp/jest_rt","clearMocks":false,"coveragePathIgnorePatterns":["/node_modules/"],"cwd":"/home/sam/gatsby-source-shopify-graphql-poc/plugin","detectLeaks":false,"detectOpenHandles":false,"errorOnDeprecated":false,"extraGlobals":[],"forceCoverageMatch":[],"globals":{},"haste":{"computeSha1":false,"throwOnModuleCollision":false},"injectGlobals":true,"moduleDirectories":["node_modules"],"moduleFileExtensions":["js","json","jsx","ts","tsx","node"],"moduleNameMapper":[],"modulePathIgnorePatterns":[],"name":"7650e90a1ff57f0cbbb2a111f8134593","prettierPath":"prettier","resetMocks":false,"resetModules":false,"restoreMocks":false,"rootDir":"/home/sam/gatsby-source-shopify-graphql-poc/plugin","roots":["/home/sam/gatsby-source-shopify-graphql-poc/plugin"],"runner":"jest-runner","setupFiles":[],"setupFilesAfterEnv":[],"skipFilter":false,"slowTestThreshold":5,"snapshotSerializers":[],"testEnvironment":"/home/sam/gatsby-source-shopify-graphql-poc/plugin/node_modules/jest-environment-node/build/index.js","testEnvironmentOptions":{},"testLocationInResults":false,"testMatch":["**/__tests__/**/*.[jt]s?(x)","**/?(*.)+(spec|test).[tj]s?(x)"],"testPathIgnorePatterns":["/node_modules/"],"testRegex":[],"testRunner":"/home/sam/gatsby-source-shopify-graphql-poc/plugin/node_modules/jest-jasmine2/build/index.js","testURL":"http://localhost","timers":"real","transform":[["^.+\\.tsx?$","/home/sam/gatsby-source-shopify-graphql-poc/plugin/node_modules/ts-jest/dist/index.js",{}]],"transformIgnorePatterns":["/node_modules/","\\.pnp\\.[^\\/]+$"],"watchPathIgnorePatterns":[]},"logLevel":20,"namespace":"backports","package":"ts-jest","version":"26.5.3"},"message":"backporting config","sequence":7,"time":"2021-03-11T16:29:23.579Z"}
{"context":{"jestConfig":{"automock":false,"cache":true,"cacheDirectory":"/tmp/jest_rt","clearMocks":false,"coveragePathIgnorePatterns":["/node_modules/"],"cwd":"/home/sam/gatsby-source-shopify-graphql-poc/plugin","detectLeaks":false,"detectOpenHandles":false,"errorOnDeprecated":false,"extraGlobals":[],"forceCoverageMatch":[],"globals":{"ts-jest":{}},"haste":{"computeSha1":false,"throwOnModuleCollision":false},"injectGlobals":true,"moduleDirectories":["node_modules"],"moduleFileExtensions":["js","json","jsx","ts","tsx","node"],"moduleNameMapper":[],"modulePathIgnorePatterns":[],"name":"7650e90a1ff57f0cbbb2a111f8134593","prettierPath":"prettier","resetMocks":false,"resetModules":false,"restoreMocks":false,"rootDir":"/home/sam/gatsby-source-shopify-graphql-poc/plugin","roots":["/home/sam/gatsby-source-shopify-graphql-poc/plugin"],"runner":"jest-runner","setupFiles":[],"setupFilesAfterEnv":[],"skipFilter":false,"slowTestThreshold":5,"snapshotSerializers":[],"testEnvironment":"/home/sam/gatsby-source-shopify-graphql-poc/plugin/node_modules/jest-environment-node/build/index.js","testEnvironmentOptions":{},"testLocationInResults":false,"testMatch":["**/__tests__/**/*.[jt]s?(x)","**/?(*.)+(spec|test).[tj]s?(x)"],"testPathIgnorePatterns":["/node_modules/"],"testRegex":[],"testRunner":"/home/sam/gatsby-source-shopify-graphql-poc/plugin/node_modules/jest-jasmine2/build/index.js","testURL":"http://localhost","timers":"real","transform":[["^.+\\.tsx?$","/home/sam/gatsby-source-shopify-graphql-poc/plugin/node_modules/ts-jest/dist/index.js",{}]],"transformIgnorePatterns":["/node_modules/","\\.pnp\\.[^\\/]+$"],"watchPathIgnorePatterns":[]},"logLevel":20,"namespace":"config","package":"ts-jest","version":"26.5.3"},"message":"normalized jest config","sequence":8,"time":"2021-03-11T16:29:23.580Z"}
{"context":{"logLevel":20,"namespace":"Importer","package":"ts-jest","requireResult":{"exists":true,"given":"typescript","path":"/home/sam/gatsby-source-shopify-graphql-poc/plugin/node_modules/typescript/lib/typescript.js"},"version":"26.5.3"},"message":"loaded module typescript","sequence":9,"time":"2021-03-11T16:29:23.580Z"}
{"context":{"logLevel":20,"namespace":"Importer","package":"ts-jest","version":"26.5.3"},"message":"patching typescript","sequence":10,"time":"2021-03-11T16:29:23.580Z"}
{"context":{"actualVersion":"4.2.3","expectedVersion":">=3.8 <5","logLevel":20,"namespace":"versions","package":"ts-jest","version":"26.5.3"},"message":"checking version of typescript: OK","sequence":11,"time":"2021-03-11T16:29:23.580Z"}
```

## envinfo

```
System:
    OS: Linux 5.8 Ubuntu 20.10 (Groovy Gorilla)
    Node version: 12.13.0

Npm packages:
    "@types/jest": "^26.0.20",
    "jest": "^26.6.3",
    "ts-jest": "^26.5.3",
    "typescript": "^4.2.3"
```

--- a/src/compiler/language-service.ts
+++ b/src/compiler/language-service.ts
@@ -1,12 +1,12 @@
 import { LogContexts, Logger, LogLevels } from 'bs-logger'
 import { existsSync, readFileSync, writeFile } from 'fs'
-import { basename, normalize, join } from 'path'
+import { basename, normalize, join, extname } from 'path'
 import memoize = require('lodash/memoize')
 import mkdirp = require('mkdirp')
 import type * as _ts from 'typescript'
 
-import { ConfigSet, TS_JEST_OUT_DIR } from '../config/config-set'
-import { LINE_FEED } from '../constants'
+import type { ConfigSet } from '../config/config-set'
+import { LINE_FEED, TS_TSX_REGEX } from '../constants'
 import type { CompilerInstance, SourceOutput } from '../types'
 import { Errors, interpolate } from '../utils/messages'
 
@@ -75,10 +75,7 @@ export const initializeLanguageServiceInstance = (configs: ConfigSet, logger: Lo
   }
   // Initialize memory cache for typescript compiler
   configs.parsedTsConfig.fileNames
-    .filter(
-      (fileName: string) =>
-        !configs.isTestFile(fileName) && !fileName.includes(configs.parsedTsConfig.options.outDir ?? TS_JEST_OUT_DIR),
-    )
+    .filter((fileName: string) => TS_TSX_REGEX.test(extname(fileName)) && !configs.isTestFile(fileName))
     .forEach((fileName: string) => {
       memoryCache.files.set(fileName, {
         version: 0,

